site/src/components/page-sidebar.tsx

```diff
@@ -373,25 +358,38 @@ export const Sidebar: VFC<SidebarProps> = ({
 
   const md = useMediaQuery(theme.breakpoints.up("md"));
 
-  const [selectedAnchorElement, setSelectedAnchorElement] =
-    useState<HTMLAnchorElement>();
-  const [selectedOffsetTop, setSelectedOffsetTop] = useState<number>();
   const [openedPages, setOpenedPages] = useState<string[]>(
     getInitialOpenedPages({ pages, asPath }),
   );
 
-  const maybeUpdateSelectedOffsetTop = useCallback(() => {
-    if (
-      selectedAnchorElement &&
-      selectedOffsetTop !== selectedAnchorElement.offsetTop
-    ) {
-      setSelectedOffsetTop(selectedAnchorElement.offsetTop);
+  const setSelectedAnchorElementTimeout = useRef<ReturnType<
+    typeof setTimeout
+  > | null>(null);
+
+  const setSelectedAnchorElement = useCallback((node: HTMLAnchorElement) => {
+    if (setSelectedAnchorElementTimeout.current) {
+      clearTimeout(setSelectedAnchorElementTimeout.current);
     }
-  }, [selectedOffsetTop, selectedAnchorElement]);
+
+    setSelectedAnchorElementTimeout.current = setTimeout(() => {
+      const parent = node.offsetParent as HTMLElement;
+      const min = parent!.scrollTop;
+      const max = min + parent!.offsetHeight - 100;
```

Comment:
Nit: since you're already casting
```suggestion
      const parent = node.offsetParent as HTMLElement;
      const min = parent.scrollTop;
      const max = min + parent.offsetHeight - 100;
```